os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -146,45 +146,43 @@ typedef struct {
 	i2s_bits_per_sample_t bits_per_sample; /*!< I2S bits per sample */
 
 	uint8_t channel_num;
-	uint8_t direction;
-
 } i2s_config_t;
 
-/* The state of the one I2S peripheral */
+/* I2S Device hardware configuration */
+struct amebasmart_i2s_config_s {
+	uint32_t i2s_mclk_pin;
+	uint32_t i2s_bclk_pin;
+	uint32_t i2s_ws_pin;
+	uint32_t i2s_sd_tx_pin;
+	uint32_t i2s_sd_rx_pin;
+
+	uint8_t i2s_idx;	/* I2S index*/
+	uint8_t rxenab : 1; /* True: RX transfers enabled */
+	uint8_t txenab : 1; /* True: TX transfers enabled */
+};
 
+/* The state of the one I2S peripheral */
 struct amebasmart_i2s_s {
 	struct i2s_dev_s dev; /* Externally visible I2S interface, must the first element!! */
 
 	i2s_t i2s_object;
-	uint32_t i2s_sclk_pin;
-	uint32_t i2s_ws_pin;
-	uint32_t i2s_sd_tx_pin;
-	uint32_t i2s_sd_rx_pin;
-	uint32_t i2s_mck_pin;
+	const struct amebasmart_i2s_config_s *config;	/* Port configuration */
 
 #if defined(I2S_HAVE_TX) && (0 < I2S_HAVE_TX)
 	uint8_t i2s_tx_buf[(I2S_DMA_PAGE_NUM + 1) * I2S_DMA_PAGE_SIZE]__attribute__((aligned(64))); /* Allocate buffer for use in TX, must I2S_DMA_PAGE_NUM+1 for zero buffer */
 	i2s_irq_handler tx_isr_handler;
 #endif
 
 #if defined(I2S_HAVE_RX) && (0 < I2S_HAVE_RX)
-	uint8_t *i2s_rx_buf;
+	uint8_t i2s_rx_buf[(I2S_DMA_PAGE_NUM + 1) * I2S_DMA_PAGE_SIZE]__attribute__((aligned(64))); /* Allocate buffer for use in RX, must I2S_DMA_PAGE_NUM+1 for zero buffer */
 	i2s_irq_handler rx_isr_handler;
 #endif
 
-	uint8_t i2s_num; /* I2S controller register base address */
-
 	i2s_err_cb_t err_cb; /* registered error callback function */
 	void *err_cb_arg;	 /* argiment to return with err cb call */
 
 	sem_t exclsem; /* Assures mutually exclusive acess to I2S */
 
-	uint8_t rxenab : 1; /* True: RX transfers enabled */
-	uint8_t txenab : 1; /* True: TX transfers enabled */
-
-	int sample_rate;			/*!< I2S sample rate */
-	int channel_num;			/*!< Number of channels */
-	int bits_per_sample;		/*!< Bits per sample */
```

Comment:
These are used in the below code, so they should not be removed.

Reply:
It is not used, the actual sampling_rate is stored in i2s_t i2s_object.

Reply:
In few places priv->sample_rate is used,

https://github.com/ziyik/TizenRT/blob/a897beb1b0ce7114a163a4c2d50e3ada7a3e6cc8/os/arch/arm/src/amebasmart/amebasmart_i2s.c#L1589C1-L1589C27

https://github.com/ziyik/TizenRT/blob/a897beb1b0ce7114a163a4c2d50e3ada7a3e6cc8/os/arch/arm/src/amebasmart/amebasmart_i2s.c#L584

If not used, then these also should be modified. This was resulting in compilation error.